libkbfs/tlf_handle_extension_test.go

```diff
@@ -0,0 +1,170 @@
+		}
+		if len(exts) != 1 {
+			t.Fatalf("Expected 1 extension, got: %d", len(exts))
+		}
+		e2 := exts[0]
+		if e2.Number != e.Number {
+			t.Fatalf("Expected %d, got: %d", e.Number, e2.Number)
+		}
+		if e2.Date != e.Date {
+			t.Fatalf("Expected %d, got: %d", e.Date, e2.Date)
+		}
+		if e2.String() != e.String() {
+			t.Fatalf("Expected %s, got: %s", e, e2)
+		}
+	}
+}
+
+func TestTlfHandleExtensionKnownTime(t *testing.T) {
+	e := &TlfHandleExtension{
+		Date:   1462838400,
+		Number: 1,
+		Type:   TlfHandleExtensionFinalized,
+	}
+	expect := "(finalized 2016-05-10)"
+	if e.String() != expect {
+		t.Fatalf("Expected %s, got: %s", expect, e)
+	}
+	e2 := &TlfHandleExtension{
+		Date:   1462838400,
+		Number: 12345,
+		Type:   TlfHandleExtensionConflict,
+	}
+	expect = "(conflicted copy 2016-05-10 #12345)"
+	if e2.String() != expect {
+		t.Fatalf("Expected %s, got: %s", expect, e2)
+	}
+}
+
+func TestTlfHandleExtensionErrors(t *testing.T) {
+	_, err := NewTlfHandleExtension(TlfHandleExtensionConflict, 0)
+	if err != ErrTlfHandleExtensionInvalidNumber {
+		t.Fatalf("Expected ErrTlfHandleExtensionInvalidNumber, got: %v", err)
+	}
+	_, err = ParseTlfHandleExtensionSuffix("(conflicted copy 2016-05-10 #0)")
+	if err != ErrTlfHandleExtensionInvalidNumber {
+		t.Fatalf("Expected ErrTlfHandleExtensionInvalidNumber, got: %v", err)
+	}
+	_, err = ParseTlfHandleExtensionSuffix("(conflicted copy 2016-05-10 #1)")
+	if err != ErrTlfHandleExtensionInvalidNumber {
+		t.Fatalf("Expected ErrTlfHandleExtensionInvalidNumber, got: %v", err)
+	}
+	_, err = ParseTlfHandleExtensionSuffix("nope")
+	if err != ErrTlfHandleExtensionInvalidString {
+		t.Fatalf("Expected ErrTlfHandleExtensionInvalidString, got: %v", err)
+	}
+	_, err = ParseTlfHandleExtensionSuffix("(conflicted copy #2)")
+	if err != ErrTlfHandleExtensionInvalidString {
+		t.Fatalf("Expected ErrTlfHandleExtensionInvalidString, got: %v", err)
+	}
+	_, err = ParseTlfHandleExtensionSuffix("(conflicted copy 2016-05-10 #)")
+	if err != ErrTlfHandleExtensionInvalidString {
+		t.Fatalf("Expected ErrTlfHandleExtensionInvalidString, got: %v", err)
+	}
+}
+
+type tlfHandleExtensionFuture struct {
+	TlfHandleExtension
+	extra
+}
+
+func (ci tlfHandleExtensionFuture) toCurrentStruct() currentStruct {
+	return ci.TlfHandleExtension
+}
+
+func TestTlfHandleExtensionUnknownFields(t *testing.T) {
+	testStructUnknownFields(t,
+		tlfHandleExtensionFuture{
+			TlfHandleExtension{
+				time.Now().UTC().Unix(),
+				2,
+				TlfHandleExtensionFinalized,
+				codec.UnknownFieldSetHandler{},
+			},
+			makeExtraOrBust("TlfHandleExtension", t),
+		})
+}
+
+func TestTlfHandleExtensionMultiple(t *testing.T) {
+	e, err := NewTlfHandleExtension(TlfHandleExtensionConflict, 1)
+	if err != nil {
+		t.Fatal(err)
+	}
+	e2, err := NewTlfHandleExtension(TlfHandleExtensionFinalized, 2)
+	if err != nil {
+		t.Fatal(err)
+	}
+	exts := []TlfHandleExtension{*e, *e2}
+	exts2, err := ParseTlfHandleExtensionSuffix(NewTlfHandleExtensionSuffix(exts))
```

Comment:
Can we please explicitly verify that `NewTlfHandleExtensionSuffix` returns the expected string?  I was expecting to see something of the form "(conflicted...) (finalized...)" among these tests and never did.

Also worth adding a test for the "(finalized...) (conflicted...)" case I mentioned above, to check whatever behavior we decide is best.